__tests__/integration/hathorwallet_facade.test.js

```diff
@@ -0,0 +1,1665 @@
+  });
+
+  it('should destroy melt authorities', async () => {
+    const hWallet = await generateWalletHelper();
+    await GenesisWalletHelper.injectFunds(hWallet.getAddressAtIndex(0), 10);
+    const { hash: tokenUid } = await createTokenHelper(
+      hWallet,
+      'Token for MeltDestroy',
+      'DMELT',
+      100
+    );
+
+    // Adding another melt authority
+    const { hash: newMeltTx } = await hWallet.delegateAuthority(
+      tokenUid,
+      'melt',
+      hWallet.getAddressAtIndex(0)
+    );
+    await waitForTxReceived(hWallet, newMeltTx);
+
+    // Validating though getMintAuthority
+    let meltAuthorities = await hWallet.getMeltAuthority(tokenUid, { many: true });
+    expect(meltAuthorities).toHaveLength(2);
+
+    // Destroying one melt authority
+    await waitUntilNextTimestamp(hWallet, newMeltTx);
+    const { hash: destroyMeltTx } = await hWallet.destroyAuthority(tokenUid, 'melt', 1);
+    await waitForTxReceived(hWallet, destroyMeltTx);
+    meltAuthorities = await hWallet.getMeltAuthority(tokenUid, { many: true });
+    expect(meltAuthorities).toHaveLength(1);
+
+    // Destroying all melt authorities
+    await waitUntilNextTimestamp(hWallet, destroyMeltTx);
+    const { hash: destroyAllMintTx } = await hWallet.destroyAuthority(tokenUid, 'melt', 1);
+    await waitForTxReceived(hWallet, destroyAllMintTx);
+    meltAuthorities = await hWallet.getMeltAuthority(tokenUid, { many: true });
+    expect(meltAuthorities).toHaveLength(0);
+
+    // Trying to melt and validating its error object
+    const meltFailure = await hWallet.meltTokens(tokenUid, 100)
+      .catch(err => err);
+    expect(meltFailure).toHaveProperty('success', false);
+    expect(meltFailure.message).toContain('authority output');
+  });
+});
+
+describe('createNFT', () => {
+  afterEach(async () => {
+    await stopAllWallets();
+    await GenesisWalletHelper.clearListeners();
+  });
+
+  it('should create an NFT with mint/melt authorities', async () => {
+    const hWallet = await generateWalletHelper();
+    await GenesisWalletHelper.injectFunds(hWallet.getAddressAtIndex(0), 100);
+
+    // Creating one NFT with default authorities
+    const nftTx = await hWallet.createNFT(
+      'New NFT',
+      'NNFT',
+      1,
+      sampleNftAddress,
+      {
+        createMint: true,
+        createMelt: true
+      },
+    );
+    expect(nftTx.hash).toBeDefined();
+    expect(nftTx.name).toEqual('New NFT');
+    expect(nftTx.symbol).toEqual('NNFT');
+    await waitForTxReceived(hWallet, nftTx.hash);
+
+    // Validating HTR fee payment
+    const htrBalance = await hWallet.getBalance(HATHOR_TOKEN_CONFIG.uid);
+    expect(htrBalance[0].balance.unlocked).toEqual(98); // 1 deposit, 1 fee
+    let nftBalance = await hWallet.getBalance(nftTx.hash);
+    expect(nftBalance[0].balance.unlocked).toEqual(1);
+
+    /*
+     * Since NFT authority tokens cannot be read by getMintAuthority, we will try to actually
```

Comment:
Why they can't? They are normal authorities. If you can't, it's an error and we should create an issue for that

Reply:
I managed to improve this test on [61793584](https://github.com/HathorNetwork/hathor-explorer-service/pull/174/commits/61793584fe01717ea7787268b72afc430cecbc49), using `getMintAuthority` and `getMeltAuthority` successfully.